packages/react-conformance/src/defaultTests.tsx

```diff
@@ -73,4 +75,89 @@ export const defaultTests: TestObject = {
+
+  /** Ensures component has top level file in package/src/componentName */
+  'has-top-level-file': (componentInfo: ComponentDoc, testInfo: TestingOptions) => {
+    if (!testInfo.isInternal) {
+      it(`has corresponding top-level file 'package/src/Component'`, () => {
+        const { displayName, componentPath, Component } = testInfo;
+        const rootPath = componentPath.replace(/src.*/, '');
+        const topLevelFile = require(path.join(rootPath, 'src', displayName));
+
+        expect(topLevelFile.default ? topLevelFile.default : topLevelFile[displayName]).toBe(Component);
+      });
+    }
+  },
+
+  /** If the component is a subcomponent, ensure its parent has the subcomponent as static property */
+  'is-static-property': (componentInfo: ComponentDoc, testInfo: TestingOptions) => {
+    it(`is a static property of its parent`, () => {
+      const { componentPath, displayName, useDefaultExport } = testInfo;
+      const componentFolder = componentPath.replace(path.basename(componentPath) + path.extname(componentPath), '');
+      const dirName = path.basename(componentFolder).replace(path.extname(componentFolder), '');
+      const isParent = displayName === dirName;
+
+      if (!isParent) {
+        const subComponentName = displayName.replace(dirName, '');
+        const subComponentFile = require(path.join(componentFolder, subComponentName));
+        const SubComponent = useDefaultExport ? subComponentFile.default : subComponentFile[subComponentName];
+
+        expect(ReactIs.isValidElementType(SubComponent)).toBe(true);
+      }
+    });
+  },
+
+  /** Ensures aria attributes are kebab cased */
+  'kebab-aria-attributes': (componentInfo: ComponentDoc, testInfo: TestingOptions) => {
+    it(`Aria attributes are kebab-cased`, () => {
+      const { Component, customMount, requiredProps } = testInfo;
+      const mountedComponent = customMount
+        ? customMount(<Component {...requiredProps} />)
+        : mount(<Component {...requiredProps} />);
+      const props = mountedComponent.props();
+
+      if (props) {
+        for (const prop of Object.keys(props)) {
+          if (prop.toLowerCase().includes('aria')) {
+            expect(prop.split(/[A-Z][a-z]+/)?.length).toBeLessThanOrEqual(1);
+            expect(prop.split('-').length).toBeGreaterThan(1);
+          }
+        }
+      }
+    });
+  },
+
+  /** Ensures that components have consistent custom callback names i.e. on[Part][Event] */
+  'consistent-callback-names': (componentInfo: ComponentDoc, testInfo: TestingOptions) => {
+    it(`has consistent custom callback names`, () => {
+      const { Component, customMount, requiredProps } = testInfo;
+      const mountedComponent = customMount
+        ? customMount(<Component {...requiredProps} />)
+        : mount(<Component {...requiredProps} />);
+      const props = mountedComponent.props();
+
+      if (props) {
+        for (const prop of Object.keys(props)) {
+          if (prop.slice(0, 3) === 'on') {
```

Comment:
This is probably not quite what we want to test. Instead I'd recommend checking `if (/^on[A-Z]/.test(propName))` and then:
1. Remove the "on" prefix
2. Get the last word. It should *not* end with "ed". (We might also want to test it against a valid list of verbs, but not sure how to get that list.)
3. The remaining words (with first letter converted back to lowercase) should probably match a prop name. 

Due to existing violations of these rules, at this point we'll probably need to introduce test-specific options: a mapping from test name to options object. Initially it would be like:
```
interface ???? {
  'consistent-callback-names'?: {
    ignoreProps?: string[];
  };
}
```
The ??? is because I think the best name for this interface would be `TestOptions`, but that's already taken right now by the interface for conformance test options all-up. So we'd need to either come up with a different name, or rename the all-up options interface to something like `IsConformantOptions` (which I think would make more sense anyway).